assembl/static/js/app/views/tokenVoteSession.js

```diff
@@ -1708,19 +1719,107 @@ var TokenVoteSessionModal = Backbone.Modal.extend({
   cancelEl: '.close, .js_close',
 
+  initialize: function(options) {
+    var that = this;
+
+    that.throttledScroll = _.throttle(that.myOnScroll, 100);
+  },
+
+  beforeCancel: function(){
+    window.location = Ctx.getDiscussionGenericURL();
+  },
+
   onShow: function(){
     var that = this;
-    var container = this.$el.find(".js_modal-body");
-    container.empty();
+
+    var displayRegion = new Marionette.Region({
+      el: that.$(".popin-body")
+    });
+
+    var outerHeader = this.$(".popin-header");
+    var displayView = new TokenVoteSessionView(
+      {
+        model: this.model,
+        showConfirmationInModal: true,
+        outerHeader: outerHeader
+      }
+    );
+
+    displayRegion.show(displayView);
+
+    that.availableTokensPositionTop = that.$(".available-tokens").position().top;
+    that.$(".available-tokens").width(that.$(".popin-body").width());
+    var popinHeaderTotalHeight = that.$(".popin-header").outerHeight();
+    that.$(".available-tokens").css("top", popinHeaderTotalHeight + (popinHeaderTotalHeight > 0 ? "px" : ""));
+    that.$(".available-tokens-container").css('min-height', "36px");
+
+    that.$(".popin-body").on('scroll', _.bind(that.onScroll, that)); // scroll event does not bubble up, and scrollable element is now .popin-body instead of this.$el
+  },
+
+  onScroll: function(){
+    this.throttledScroll();
+  },
+
+  myOnScroll: function(){
+    if (this.$(".popin-body").scrollTop() > this.availableTokensPositionTop) {
+      this.$(".available-tokens").addClass("fixed");
+    }
+    else {
+      this.$(".available-tokens").removeClass("fixed");
+    }
+  },
+
+  serializeData: function() {
+    return {
+      modal_title: i18n.gettext("Collective votes"),
+    };
+  },
+
+  onDestroy: function(){
+    if ( this.model ){
+      /*
+      We re-fetch widget data from the server, so that when the user opens the vote popin again,
+      up-to-date information is displayed (because this widget data comes from a model contained
+      in the collection collectionManager::getAllWidgetsPromise()).
+      TODO: This is a bit hacky, maybe we should find a better way, for example when popin opens, re-load user votes (instead of widget) and use these instead of the ones listed in the widget.
+      */
+      this.model.fetch();
+    }
+    Ctx.clearModal({destroyModal: false});
+  }
+});
+
+var TokenVoteSessionSubmittedView = Marionette.LayoutView.extend({
+  constructor: function TokenVoteSessionSubmittedView(){
+    Marionette.LayoutView.apply(this, arguments);
+  },
+
+  template: false,
+  className: 'token-vote-session-submitted',
+
+  initialize(options){
+    this.editInModal = ("editInModal" in options && options.editInModal == true) ? true : false;
+    console.log("TokenVoteSessionSubmittedView::initialize() editInModal:", this.editInModal);
+  },
+
+  onShow: function(){
+    var that = this;
+    console.log("TokenVoteSessionSubmittedView::onShow() editInModal:", that.editInModal);
```

Comment:
Should put all these `console.log's in a wrapped `Ctx.debugRender`, if you need them at all.